pkg/prometheus/prometheus.go

```diff
@@ -80,6 +82,52 @@ func getRouter(registry Registry, path string) (*http.ServeMux, error) {
 	return mux, nil
 }
 
+// GetRegistry returns a fully configured Prometheus registry with metrics based on Kube Linter validations
+func GetRegistry() (*prometheus.Registry, error) {
+	prom := prometheus.NewRegistry()
+
+	reg, err := validations.GetKubeLinterRegistry()
+	if err != nil {
+		return nil, err
+	}
+
+	checks, err := validations.GetAllNamesFromRegistry(reg)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, checkName := range checks {
+		metric, err := setupMetric(reg, checkName)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create metric for check %s", checkName)
```

Comment:
Hmm so I guess this means that if the `checkName` is not known (or maybe just mispelled. I am looking at the line 115 below) then the DVO init will fail with error right? Do we want such behavior?

Reply:
This was an existing logic within the validationEngine package where an error on spelling a validation name will throw an error.

I agree this could be handled better with a logline and skipping the error. I'll do some testing and come back with a change as I'm refactoring the function where this behaviour was originally.

Reply:
Yes you're right, but previously the values were hardcoded,  but now they can be changed as the user wishes, which means it is more error-prone in my opinion.

Reply:
Yes, good point. I'll do some testing and refactor this with a different approach.
Since we can't block ConfigMap update, I'll validate the structure (currently just marshalling into a struct) and then allow errors in the own configuration (logline informing about the wrong spelling/configuration and skipping any errors that could break the system.

Reply:
After reviewing this and testing it, this will only trigger an error if the validation checkname is wrong at software start.

If this comes from an update on the ConfigMap, the error will be ignored and logged by the code.

this line :point_down: 
https://github.com/app-sre/deployment-validation-operator/pull/263/files#diff-48a7386f3f126ad7b562cfde8d8b4ac24bf7e080225fc7afeee7b35eb2ed3161R198